src/post/schemas/post.schema.ts

```diff
@@ -0,0 +1,46 @@
+import { HydratedDocument } from 'mongoose';
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+
+export type PostDocument = HydratedDocument<Post>;
+
+@Schema()
+export class Post {
+  @Prop({ required: true, unique: true })
+  postId: string;
+
+  @Prop({ required: true })
+  title: string;
+
+  @Prop({ required: true })
+  imageUrl: string;
+
+  @Prop({ required: true })
+  description: string;
+
+  @Prop({ required: true })
+  tag: [string];
+
+  @Prop({ required: true})
+  points: number;
+
+  @Prop({ required: true })
+  userId: string;
+
+  @Prop({required: true })
+  numOfLike: number;
+
+  @Prop ({required: true })
+  postedDate: Date;
+
+  @Prop ()
+  dueDate: Date;
+
+  @Prop ({required: true })
+  completed: boolean;
```

Comment:
I guess `dueDate` means the date of the event date?
If there's a `dueDate`, then if now > `dueDate` means it's completed? If so, I am guessing maybe `completed` is not necessary?

Reply:
in some situations, the user may want to close the post before the due date, so I think adding a completed option is reasonable?

Reply:
I see, maybe rename `dueDate` to `eventDate`.
For `completed`, I think it's hard to reflect on the different conditions of the event, like open for registration, cancelled, completed, postponed, ... maybe use an enum `status` and the default is `open`?

Reply:
updated

Reply:
update